components/brave_wallet/browser/cardano/cardano_knapsack_solver_unittest.cc

```diff
@@ -166,15 +165,12 @@ TEST_F(CardanoKnapsackSolverUnitTest, NoChangeGenerated) {
     inputs.push_back(MakeMockTxInput(total_input, 0));
     CardanoKnapsackSolver solver(base_tx, latest_epoch_parameters(), inputs);
     auto tx = solver.Solve();
-    ASSERT_TRUE(tx.has_value());
 
-    // We have a bit more than send amount + fee. Still no change. Surplus goes
-    // to fee.
-    EXPECT_EQ(tx->EffectiveFeeAmount(), min_fee + 1);
-    EXPECT_EQ(tx->TotalInputsAmount(), total_input);
-    EXPECT_EQ(tx->TotalOutputsAmount(), send_amount());
-    EXPECT_EQ(tx->TargetOutput()->amount, send_amount());
-    EXPECT_FALSE(tx->ChangeOutput());
+    // We have a bit more than send amount - min_fee. Can't create transaction
```

Comment:
This is an odd edge case that they introduced with their fee structure. Thanks for keeping the test around for this. I agree it makes sense to just error out in this case.